src/api/ApiHelper.tsx

```diff
@@ -494,14 +493,13 @@ function initAPI(): API {
             let requestData = {
                 name: itemTagOrName,
                 start: Math.round(fetchStart / 100000) * 100,
-                reforge: itemFilter?.reforge ? itemFilter.reforge.id : undefined,
-                enchantments: itemFilter && itemFilter.enchantment !== undefined && itemFilter.enchantment.level !== undefined && itemFilter.enchantment.id ? [[itemFilter.enchantment.id, itemFilter.enchantment.level]] : undefined
+                filter:itemFilter
             };
             httpApi.sendLimitedCacheRequest({
                 type: RequestType.RECENT_AUCTIONS,
                 data: requestData,
                 resolve: (data: any) => {
-                    resolve(data.map(a => parseRecentAuction(a)));
+                    resolve(data?.map(a => parseRecentAuction(a)));
```

Comment:
why the ?
shouldt it be save to assume a array? If not check for a empty response and return []

`if(!data){return [])`